### ISSUE TITLE
Memory leak on Linux

### ISSUE THREAD

**Issue by evgeniy-harchenko** [AvaloniaUI/Avalonia#15008]
### Describe the bug

Memory allocates every time while interacting with any UI elements. It doesn't deallocate at all. It seems GC is not running at all.
Temporary solution: periodically force run `GC.Collect()`.

### To Reproduce

1. Create sample project.
2. Run the project.
3. Open memory profiler.
4. Move mouse cursor over any UI elements.

### Expected behavior

Like on Windows: without memory leak.

### Avalonia version

11.0.10-11.0.0 (other not tried)

### OS

Linux

### Additional context

Tested on KDE neon and Manjaro with KDE Plasma 5.27.10 (X11)

<details>
  <summary>Example GIF</summary>

![leak](https://github.com/AvaloniaUI/Avalonia/assets/42656259/14bb17c2-ef8b-4485-a0ca-8613e512308f)
</details>

**Comment by maxkatz6** [comment]
> Temporary solution: periodically force run GC.Collect().

If GC.Collect works, then it's not a leak.

**Comment by thevortexcloud** [comment]
Indeed, the garbage collector will run _eventually_. When it runs though is entirely up to the GC.

**Comment by SydneyOwl** [comment]
I am using Avalonia 12 and have reproduced the same issue on Linux (memory continuously grows and never reclaimed after repeatedly opening/closing a specific page). The same application runs fine on Windows 11 with stable memory usage. I profiled it on Ubuntu 24 and verified that views and ViewModels are properly collected by the GC. dotnet-gcdump and counters indicate that the managed heap remains small. The RSS increase appears to originate from SkiaSharp's native allocator rather than the managed heap, and seems like glibc's allocator does not promptly return freed pages to the OS as well, resulting in persistently high memory usage.

As a temporary solution I called malloc_trim(0) via P/Invoke after window closure to actively release free pages back to the OS. With this workaround in place, memory usage drops significantly, and RSS can be observed to fall back after every page close.

```csharp

internal static class xxxxxxx
{
    private static readonly SemaphoreSlim TrimLock = new(1, 1);

    public static void TrimAfterWindowClosed()
    {
        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) return;

        _ = Task.Run(async () =>
        {
            if (!await TrimLock.WaitAsync(0).ConfigureAwait(false)) return;

            try
            {
                await Task.Delay(500).ConfigureAwait(false);

                // optional; for test purpose
                GCSettings.LargeObjectHeapCompactionMode = GCLargeObjectHeapCompactionMode.CompactOnce;
                GC.Collect(GC.MaxGeneration, GCCollectionMode.Forced, true, true);
                GC.WaitForPendingFinalizers();
                GC.Collect(GC.MaxGeneration, GCCollectionMode.Forced, true, true);

                var trimmed = false;
                try
                {
                    trimmed = malloc_trim(UIntPtr.Zero) != 0;
                }
                catch (Exception ex) when (ex is DllNotFoundException or EntryPointNotFoundException)
                {
                    ClassLogger.Trace(ex, "malloc_trim unavailable on this libc.");
                }
            }
            catch (Exception ex)
            {
                ClassLogger.Debug(ex, "memory trim failed.");
            }
            finally
            {
                TrimLock.Release();
            }
        });
    }

    [DllImport("libc", EntryPoint = "malloc_trim", SetLastError = false)]
    private static extern int malloc_trim(UIntPtr pad);
}

```
